tests/container_test.go

```diff
@@ -560,3 +561,48 @@ func requireEstimationsMatch(t *testing.T, expected []estimation, actual []estim
 		require.True(t, found, "expected estimation from %x to be present", e.from)
 	}
 }
+
+func TestContainerList(t *testing.T) {
+	c, _, _ := newContainerInvoker(t, false)
+
+	cID := cidtest.ID()
+	var containerList []any
+	for range 10 {
+		containerList = append(containerList, dummyNodeInfo(c.NewAccount(t)).raw)
+	}
+
+	c.Invoke(t, stackitem.Null{}, "addNextEpochNodes", cID[:], containerList)
+	c.Invoke(t, stackitem.Null{}, "commitContainerListUpdate", cID[:])
+
+	stack, err := c.TestInvoke(t, "nodes", cID[:])
+	require.NoError(t, err)
+
+	iter, ok := stack.Pop().Value().(*storage.Iterator)
+	require.True(t, ok)
+
+	fromContract := iteratorToBytesArray(t, iter)
+	require.ElementsMatch(t, containerList, fromContract)
+
+	// clean up container
+	c.Invoke(t, stackitem.Null{}, "commitContainerListUpdate", cID[:])
```

Comment:
1st i'd make one more `addNextEpochNodes` call with `containerList` subset + several completely new elements. And check that old elements are dropped. This is quite non-trivial with add-add-add... semantic sequence